lib/analyse_ocamlformat.ml

```diff
@@ -66,12 +66,23 @@ let ocamlformat_version_from_file ~root job path =
     | _::_::_ ->
         Lwt.return (Error (`Msg "Multiple 'version=' lines in .ocamlformat"))
 
-let get_ocamlformat_source job ~opam_files ~root =
+let get_ocamlformat_source job ~opam_files ~root ~find_opam_repo_commit =
+  let ( let* ) = Lwt_result.Infix.( >>= ) in
   let proj_is_ocamlformat p = String.equal (Filename.basename p) "ocamlformat.opam" in
   match List.find_opt proj_is_ocamlformat opam_files with
   | Some opam_file ->
     let path = Filename.dirname opam_file in
-    Lwt_result.return (Some (Vendored { path }))
+    Lwt_result.return (Some (Vendored { path }), None)
   | None ->
-    Fpath.(to_string (root / ".ocamlformat")) |> ocamlformat_version_from_file ~root job
-    |> Lwt_result.map (Option.map (fun version -> Opam { version }))
+    let* version_in_dot_ocamlformat =
+      ocamlformat_version_from_file ~root job
+        Fpath.(to_string (root / ".ocamlformat"))
+    in
+    match version_in_dot_ocamlformat with
+    | None ->
+            Lwt_result.return (None, None)
+            (* This case happens when the ocamlformat file doesn't specify a version.
+               It will try to get the latest version for the last platform. *)
```

Comment:
Being opinionated here, this should probably fail or suggest adding the ocamlformat version to the file as a warning before trying the latest ocamlformat version. https://github.com/ocaml-ppx/ocamlformat#formatting-code

If someone hasn't specified a version, are they expecting to follow the latest or have they neglected to add it and need to be gently reminded.